### PR TITLE
Ssza: Initial draft implementation

### PR DESCRIPTION
**Pull request by pstuecker**

### CODE REVIEW

File: java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformator.xtend [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+						filterNull.toList
+
+					val pos = (sue.filter(Zugeinwirkung).
+						map[punktObjektStrecke] + sue.filter(FMA_Anlage).flatMap [
+						val fma = it
+						container.FMAKomponente.filter [
+							IDFMAgrenze.contains(fma)
+						].map[punktObjektStrecke]
+					] + sue.filter(FMA_Komponente).map[punktObjektStrecke]).
+						flatten
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformator.xtend:211]:
```suggestion
					val pos = #[]
					sue.forEach [
						switch (it) {
							Zugeinwirkung,
							FMA_Komponente:
								pos.addAll(punktObjektStrecke)
							FMA_Anlage: {
								val fmaKomponente = container.FMAKomponente.filter[IDFMAgrenze.contains(it)]
								pos.addAll(fmaKomponente.flatMap[punktObjektStrecke])		
							}	
						}
					]
```
here could be use switch in for loop too avoid loop _sue_ 3 times and easier to understand

File: java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformator.xtend [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+				[
+					val sue = schaltmittelZuordnung.map[IDSchalter?.value]?.
+						filterNull.toList
+
+					val pos = (sue.filter(Zugeinwirkung) +
+						sue.filter(FMA_Komponente)).flatMap [
+						getKMStrecke.apply(it)
+					]
+					// TODO FMA_Anlage
+					return pos
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformator.xtend:256]:
```suggestion
					return sue.flatMap[
						switch(it) {
							Zugeinwirkung,
							FMA_Komponente:
								return getKMStrecke.apply(it)
							// TODO FMA_Anlage
						}
						return null
					].filterNull
```
same here

File: java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformator.xtend [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+		fill(
+			cols.getColumn(Bemerkung),
+			datenpunkt,
+			[
+				"TODO" // TODO
+			]
+		)
+
+		// Y: Sslb.Bemerkung
+		fillFootnotes(datenpunkt)
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformator.xtend:371]:
```suggestion
		// O: Ssza.Bemerkung
		fillFootnotes(datenpunkt)
```
duplicated

**Review by TruongQuangSB — CHANGES_REQUESTED** [review]

File: java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformationService.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ * 
+ * @author rumpf
+ * 
+ * @usage production
+ */
+@Component(service = {
+		PlanPro2TableTransformationService.class }, immediate = true, property = {
+				"table.shortcut=ssza" })
+public final class SszaTransformationService
+		extends AbstractPlanPro2TableTransformationService {
```
[the remark this answers is not in the document]
Comment (pstuecker) [reply java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformationService.java:34]: the provided sorting priority does not make any sense

**Review by TruongQuangSB — APPROVED** [review]

**Review by TruongQuangSB — APPROVED** [review]

File: java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformator.xtend [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+	var TMFactory factory = null
+
+	new(Set<ColumnDescriptor> cols,
+		EnumTranslationService enumTranslationService,
+		TopologicalGraphService topGraphService) {
+		super(cols, enumTranslationService)
+	}
+
+	override transformTableContent(MultiContainer_AttributeGroup container,
+		TMFactory factory) {
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.feature.table.pt1/src/org/eclipse/set/feature/table/pt1/ssza/SszaTransformator.xtend:58]:
```suggestion
	override transformTableContent(MultiContainer_AttributeGroup container,
		TMFactory factory, Stell_Bereich controlArea) {
```

**Review by TruongQuangSB — CHANGES_REQUESTED** [review]
Build error

**Review by TruongQuangSB — APPROVED** [review]

**Pull request merged**